tests/new_unit_tests/Commands/Connection/ConnectionCommandTest.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * @file
+ *
+ * Contains Pantheon\Terminus\UnitTests\Commands\Connection\ConnectionCommandTest
+ */
+
+namespace Pantheon\Terminus\UnitTests\Commands\Connection;
+
+
+use ReflectionMethod;
+
+abstract class ConnectionCommandTest extends \PHPUnit_Framework_TestCase
```

Comment:
I think that key failure cases should be tested.

As for protected methods, ideally each class has a public API (set of public methods) that is tested fully, and that fully exercises the protected methods.  n.b. the public methods of internal classes should be tested too.

Sometimes protected methods should be tested directly. It depends on how much complexity is encapsulated in the protected methods.  Using a protected method caller is preferable to making a test double and asserting that certain methods are called, IMO.